samples/graph-activity-feed-broadcast/csharp/README.md

```diff
@@ -0,0 +1,142 @@
+11. In the **Authorized client applications** section, identify the applications that you want to authorize for your app’s web application. Each of the following IDs needs to be entered:
+    * `1fec8e78-bce4-4aaf-ab1b-5451cc387264` (Teams mobile/desktop application)
+    * `5e3ce6c0-2b1f-4285-8d4b-75ee78787346` (Teams web application)
+12. Navigate to **API Permissions**, and make sure to add the follow permissions:
+-   Select Add a permission
+-   Select Microsoft Graph -\> Delegated permissions.
+    - `User.Read` (enabled by default)
+    - `Directory.Read.All`
+    - `Directory.ReadWrite.All`
+	- `TeamsAppInstallation.ReadForUser`
+	- `TeamsAppInstallation.ReadWriteForUser`
+    - `TeamsActivity.Send`
+    - `TeamsAppInstallation.ReadForUser.All`.
+
+**Note** Your need to add `TeamsActivity.Send` and `Directory.Read.All` as Application level permissions too.
+
+-   Click on Add permissions. Please make sure to grant the admin consent for the required permissions.
+13. Navigate to **Authentication**
+    If an app hasn't been granted IT admin consent, users will have to provide consent the first time they use an app.
+    Set a redirect URI:
+    * Select **Add a platform**.
+    * Select **web**.
+    * Enter the **redirect URI** for the app in the following format: `https://{Base_Url}/Auth/end`, `https://{Base_Url}/Auth/Start`. This will be the page where a successful implicit grant flow will redirect the user.
+    Enable implicit grant by checking the following boxes:  
+    ✔ ID Token  
+    ✔ Access Token  
+14.  Navigate to the **Certificates & secrets**. In the Client secrets section, click on "+ New client secret". Add a description(Name of the secret) for the secret and select “Never” for Expires. Click "Add". Once the client secret is created, copy its value, it need to be placed in the appsettings.json.
+
+15. Clone the repository
+   ```bash
+   git clone https://github.com/OfficeDev/Microsoft-Teams-Samples.git
+   ```
+
+16. Open the code in Visual Studio
+   - File -> Open -> Project/Solution
+   - Navigate to folder where repository is cloned then `samples/graph-activity-feed-broadcast/csharp/ActivityFeedBroadcast.sln`
+    
+17. Run ngrok - point to port 3978
+
+    ```bash
+    # ngrok http -host-header=rewrite 3978
+    ```
+ 
+18. Setup and run the bot from Visual Studio: 
+   Modify the `appsettings.json` and fill in the following details:
+   - `MicrosoftAppId` - Generated from Step 3 (Application (client) ID)is the application app id
+   - `TenantId` - Generated from Step 3(Directory (tenant) ID) is the tenant id
+   - `MicrosoftAppPassword` - Generated from Step 14, also referred to as Client secret
+   - `{Base_URL}` - Your application's base url. E.g. https://12345.ngrok.io if you are using ngrok.
+   - Press `F5` to run the project
+	 
+19. Modify the `manifest.json` in the `/AppPackage` folder and replace the following details:
+   - `{{Microsoft-App-Id}}` with Application id generated from Step 3
```

Comment:
cross check if the keys are used in same manner in manifest.json file e.g. spelling for Microsoft-App-Id